components/Form/FormQuantitySpecification/index.test.js

```diff
@@ -0,0 +1,30 @@
+import FormQuantitySpecification from ".";
+import { render, screen, act } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+test("When the user presses the submit-Button, the result text is rendered", async () => {
+  const user = userEvent.setup();
+  render(
+    <FormQuantitySpecification
+      foodCategory={{
+        id: "1",
+        name: "Vollkorngetreide",
+        recommendedConsumption: 232,
+        recommendedExample:
+          "2 Tassen Vollkornreis und 1,5 Scheiben Vollkornroggenbrot",
+        maxRange: 464,
+        maxRangeInputField: 600,
+      }}
+    />
+  );
+
+  const submitButton = screen.getByRole("button", {
+    name: "Auswertung",
+  });
+
+  await act(async () => {
+    await user.click(submitButton);
+  });
```

Comment:
This should work without `act` in the current version of testing library you have used in your project.